pkg/sdk/api/v1beta1/node_agent_types.go

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"github.com/banzaicloud/operator-tools/pkg/typeoverride"
+	"github.com/banzaicloud/operator-tools/pkg/types"
+	"github.com/banzaicloud/operator-tools/pkg/volume"
+)
+
+// +kubebuilder:object:generate=true
+
+type NodeAgent struct {
+	Name          string              `json:"name,omitempty"`
+	Type          string              `json:"type,omitempty"`
```

Comment:
We must emphasize here and in the docs that based on the given type, the operator will automatically configure the required nodeselectors and tolerations for the workload.

We need to highlight the fact that a fully custom configuration is also supported.